instrumentation/jsf/jsf-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jsf/JsfTracer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jsf;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import javax.faces.FacesException;
+import javax.faces.component.ActionSource2;
+import javax.faces.component.UIViewRoot;
+import javax.faces.context.FacesContext;
+import javax.faces.el.EvaluationException;
+import javax.faces.el.MethodNotFoundException;
+import javax.faces.event.ActionEvent;
+
+public abstract class JsfTracer extends BaseTracer {
+
+  private static String getSpanName(String expressionString) {
+    String result = expressionString;
+    if (result.startsWith("#{")) {
+      result = result.substring(2);
+    }
+
+    if (result.endsWith("}")) {
+      result = result.substring(0, result.length() - 1);
+    }
+
+    if (result.contains("(") && result.contains(")")) {
+      result = result.replaceAll("\\(.*?\\)", "()");
+    }
+
+    return result;
+  }
+
+  public Span startSpan(ActionEvent event) {
+    if (event.getComponent() instanceof ActionSource2) {
```

Comment:
Would be nice to have some comments about this particular constraint and how large portion of other components are we missing and why?

Reply:
Comment added. My understanding is that there aren't that many components that produce actions, those that do extend https://docs.oracle.com/javaee/7/api/javax/faces/component/UICommand.html which implements ActionSource2.